src/main/java/io/xdb/core/command/CommandRequest.java

```diff
@@ -0,0 +1,53 @@
+package io.xdb.core.command;
+
+import io.xdb.core.communication.Communication;
+import io.xdb.core.persistence.Persistence;
+import io.xdb.gen.models.CommandResults;
+import io.xdb.gen.models.CommandWaitingType;
+import io.xdb.gen.models.Context;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class CommandRequest {
+
+    private final CommandWaitingType waitingType;
+    private final List<BaseCommand> commands;
+
+    /**
+     * EMPTY command request will trigger {{@link io.xdb.core.state.AsyncState#execute(Context, Object, CommandResults, Persistence, Communication)}} immediately.
+     */
+    public static CommandRequest EMPTY = CommandRequest.builder().waitingType(CommandWaitingType.EMPTYCOMMAND).build();
+
+    /**
+     * To trigger {{@link io.xdb.core.state.AsyncState#execute(Context, Object, CommandResults, Persistence, Communication)}} after ANY of the commands has completed.
+     *
+     * @param commands  a list of commands to wait
+     * @return
+     */
+    public static CommandRequest anyCommandComplete(final BaseCommand... commands) {
```

Comment:
How about just `anyOf` and `AllOf`? 

It seems would be more natural/fluent to read than this longer name, without ambiguous